### PR TITLE
Adding new directory structure for edges

### LINKED ISSUE #69: Remove old edges.yaml files, replace with new calculated ones
As discussed with @ijpulidos and @dotsdl, we will remove the current edges.yaml files and add in a new 03_edges with an entry for a minimum spanning graph using OpenFE + the Perses mapper. We'll also eventually add one generated via OpenFE + Lomap, but that's probably going to be separate from this issue.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #69

### CODE REVIEW

File: data/cdk2/03_edges/01_perses_openfe.yaml
Comment (IAlibay) [inline data/cdk2/03_edges/01_perses_openfe.yaml:8]: Somehow the mapper created string, fix this.

**Comment by ijpulidos** [comment]
While trying to run all the edges in the data set I detected the current issues in some of the targets
```
Target pfkfb3 references non-existing ligand in edge edge_lig_38_lig_36.
Target pfkfb3 references non-existing ligand in edge edge_lig_54_lig_38.
Target pfkfb3 references non-existing ligand in edge edge_lig_46_lig_24.
Target pfkfb3 references non-existing ligand in edge edge_lig_59_lig_30.
Target pfkfb3 references non-existing ligand in edge edge_lig_53_lig_38.
Target pfkfb3 references non-existing ligand in edge edge_lig_67_lig_24.
Target thrombin references non-existing ligand in edge edge_lig_7a_lig_5.
Target cdk8 references non-existing ligand in edge edge_lig_14_lig_23.
Target cdk8 references non-existing ligand in edge edge_lig_17_lig_23.
Target cdk8 references non-existing ligand in edge edge_lig_42_lig_23.
Target cdk8 references non-existing ligand in edge edge_lig_45_lig_23.
Target cdk8 references non-existing ligand in edge edge_lig_43_lig_23.
Target cdk8 references non-existing ligand in edge edge_lig_44_lig_23.
Target cdk8 references non-existing ligand in edge edge_lig_18_lig_23.
Target p38 references non-existing ligand in edge edge_lig_p38a_3fmk_lig_p38a_2m.
Target tyk2 references non-existing ligand in edge edge_lig_jmc_30_lig_ejm_48.
Target shp2 references non-existing ligand in edge edge_lig_E14_lig_E26.
```

It's not specially easy from my script which ligands exactly are missing or being referenced in the edges but not in the `ligands.sdf`. But by just inspection I could check that for `tyk2` the ligand `lig_jmc_30` doesn't exist in https://github.com/openforcefield/protein-ligand-benchmark/blob/issues_75_and_81/data/tyk2/02_ligands/ligands.sdf (this is the revision for the latest changes with fixes).

Hence it does seem that there are some edges referencing ligands that do not exist in the latest revision for https://github.com/openforcefield/protein-ligand-benchmark/pull/82 . Now, I see some `ligands.sdf` were changed in that PR, so it might be that the network of transformations just needs to be updated with the changes on that branch/revision.

**Comment by IAlibay** [comment]
@ijpulidos - Am I correct in understanding you're using the new graph edges from this PR but applying it to the branch from #82?

If that's the case then it's partly expected, once #82 is merged then I'll re-generate the edges here to match that (I didn't want to make this depend on the other branch if we're going to squash merge things).

**Comment by ijpulidos** [comment]
> @ijpulidos - Am I correct in understanding you're using the new graph edges from this PR but applying it to the branch from #82?

Yes, I merged both branches because I needed the changes in both for my test to make sense. 

> If that's the case then it's partly expected, once #82 is merged then I'll re-generate the edges here to match that (I didn't want to make this depend on the other branch if we're going to squash merge things).

Alright! That sounds good to me. My test will still check things run on perses, so this is not a blocking issue. I just wanted to mention it in case we weren't expecting it. All good!

**Comment by ijpulidos** [comment]
Actually, on a second thought, I realize that we do need the branches to be merged for me to be able to test this correctly, since the data for the edges and the ligands is correlated.

**Comment by dotsdl** [comment]
Ready for review!

**Review by ijpulidos — COMMENTED** [review]
I could run all the edges for all the targets with perses :partying_face: 

The only thing that I saw it was curious was that the `tyk2` network of transformations is a starmap. As we discussed in the meeting, we believe this shouldn't be happening. @IAlibay can you double check that? Probably not only for `tyk2` but for all of the targets.

Otherwise, this is looking great!

**Comment by dotsdl** [comment]
@IAlibay please merge when satisfied! Thanks!

**Comment by dotsdl** [comment]
From @IAlibay : not getting minimum-spanning graphs from `perses`, but starmaps instead. Can't use latest `perses` since it doesn't have `openff-toolkit` 0.11.0 support.

**Comment by IAlibay** [comment]
> From @IAlibay : not getting minimum-spanning graphs from `perses`, but starmaps instead. Can't use latest `perses` since it doesn't have `openff-toolkit` 0.11.0 support.

On my side of things I also need to work out what was happening with the 0.4 version of openfe (which works with off-tk 0.10.x), e.g. if it's an issue about how we're ingesting perses score.

**Comment by dotsdl** [comment]
Blocked by: https://github.com/choderalab/perses/pull/1128

**Comment by ijpulidos** [comment]
Just FYI. From today's meeting I would be trying to merge the directory structure from this PR to the main branch, such that we can continue working on the transformation networks with the correct directory structure.

**Comment by ijpulidos** [comment]
The idea is that we would be dropping the current edges yaml file in this PR and just leave the directories for @IAlibay and other to build on top of it from the main branch.

**Review by ijpulidos — APPROVED** [review]
With this we should have the directory structure that we can use to built on top of it. LGTM!

**Comment by ijpulidos** [comment]
Actually, I think I prefer to have some edges files merged, even if they are really bad ones. Because that way we can still run the simulations against this branch or the main one with he same scripts, and it would also give us placeholders that we can use for future PRs, whereas if we leave the directories empty they won't even appear in the repo. I am going to revert the latest commit for now, such that we can still use this branch for some benchmarks if needed. @IAlibay thoughts?

**Comment by IAlibay** [comment]
> Actually, I think I prefer to have some edges files merged, even if they are really bad ones. Because that way we can still run the simulations against this branch or the main one with he same scripts, and it would also give us placeholders that we can use for future PRs, whereas if we leave the directories empty they won't even appear in the repo. I am going to revert the latest commit for now, such that we can still use this branch for some benchmarks if needed. @IAlibay thoughts? 

Yeah sure, I think that makes sense.

**Pull request merged**
